addons/scaffolding/plugins/fr.gouv.mindef.safran.database.ui/src/fr/gouv/mindef/safran/database/ui/actions/AbstractExportAsAction.java

```diff
@@ -0,0 +1,314 @@
+			PlatformUI.getWorkbench().getProgressService().run(true, false, operation);
+		} catch (Exception e) {
+			MessageDialog.openError(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell(), "SQL Generation", "A problem occured during the generation. See Error Log view for more details.");
+			Activator.getDefault().getLog().log( new Status(IStatus.ERROR, Activator.PLUGIN_ID, e.getMessage(), e));
+		}
+		
+	}
+	
+	protected abstract IStatus doGeneratedScripts(final Comparison comparison, final File targetFolder) throws IOException;
+	
+	private IResource getContainingFolder(Comparison comparison) {
+		if (comparison.getMatches() != null && comparison.getMatches().isEmpty() == false) {
+			Match match = comparison.getMatches().get(0);
+			Resource resource = match.getLeft().eResource();
+			if (resource instanceof CDOResource) {
+				return getModelingProject(resource);
+			} else if(resource.getURI().isPlatformResource()) {
+				String uri = resource.getURI().toPlatformString(true);
+				Path path = new Path(uri);
+				return ResourcesPlugin.getWorkspace().getRoot().getFile(path).getParent();
+			} else if(activeEditor.getEditorInput() instanceof ThreeWayResourceCompareInput) {
+				// The resource is a SVN resource
+				try {
+					IEditorInput editorInput = activeEditor.getEditorInput();
+					Field localField = editorInput.getClass().getDeclaredField("local");
+					localField.setAccessible(true);
+					ILocalResource local = (ILocalResource) localField.get(editorInput);
+					return local.getResource().getParent();
+				} catch (ReflectiveOperationException e) {
+					// The fallback case below will apply
+				} catch (SecurityException e) {
+					// The fallback case below will apply
+				} catch (IllegalArgumentException e) {
+					// The fallback case below will apply
+				}
+			}
+			
+			// Fallback case
+			
+			ContainerSelectionDialog projectSelectionDialog = new ContainerSelectionDialog(
+					activeEditor.getSite().getShell(), null, false, "Sélectionner le projet de destination :");
+			
+			projectSelectionDialog.setTitle("Sélection de projet");
+
+			if(projectSelectionDialog.open() == ContainerSelectionDialog.OK && projectSelectionDialog.getResult().length == 1) {
+				Path projectPath = (Path) projectSelectionDialog.getResult()[0];
+				IResource selectedResource = ResourcesPlugin.getWorkspace().getRoot().findMember(projectPath);
+				if(selectedResource instanceof IProject) {
+					return selectedResource;
+				}
+			}
+			
+		}
+		return null; 
+	}
+	
+	private IResource getModelingProject(Resource resource) {
+		Session session = SessionManager.INSTANCE.getSession(resource);
+		if (session != null) {
+			Resource airdResource = session.getSessionResource();
+			String uri = airdResource.getURI().toPlatformString(true);
+			Path path = new Path(uri);
+			return ResourcesPlugin.getWorkspace().getRoot().getFile(path).getParent();
+		}
+		return null;
+	}
+	
+	private File getTargetfolder(IResource containingFolder) {
```

Comment:
I'm not sure using the java.io.File for file creation triggers a workspace refresh allowing the user to see the created file. But if that code works it's okay :)

Reply:
Yes it's working because it is done afterwards.

```java
try {
					containingFolder.getProject().refreshLocal(IResource.DEPTH_INFINITE, monitor);
				} catch (CoreException e) {
					IStatus status = Activator.createErrorStatus("Error during refreshing tree fodler", e);
					Activator.getDefault().getLog().log(status);
				}
```

Reply:
Ok, thanks for pointing it to me.